zio-cli/shared/src/main/scala/zio/cli/CliApp.scala

```diff
@@ -56,6 +57,75 @@ object CliApp {
     def printDocs(helpDoc: HelpDoc): UIO[Unit] =
       printLine(helpDoc.toPlaintext(80)).!
 
+    def checkAndGetOptionsFilePaths(topLevelCommand: String): List[String] = {
+      val filename = s".$topLevelCommand"
+
+      val cwd        = java.lang.System.getProperty("user.dir")
+      val homeDirOpt = java.lang.System.getProperty("user.home")
+
+      def parentPaths(path: String): List[String] = {
+        val parts = path.split(java.io.File.separatorChar).filterNot(_.isEmpty)
+
+        (0 to parts.length)
+          .map(i =>
+            s"${java.io.File.separatorChar}${parts
+                .take(i)
+                .mkString(java.io.File.separator)}"
+          )
+          .toList
+      }
+
+      val paths        = parentPaths(cwd)
+      val pathsToCheck = homeDirOpt :: paths
+
+      val existingFiles = pathsToCheck.filter { path =>
+        val filePath = Paths.get(path, filename)
+        Files.exists(filePath)
+      }
+
+      // Print out the paths with existing files
+      existingFiles.foreach(println)
+
+      return existingFiles
+    }
+
+    def loadOptionsFromFile(
+      topLevelCommand: String
+    ): List[String] = {
+      // Get a list of file paths based on the top-level command
+      val filePaths = checkAndGetOptionsFilePaths(topLevelCommand)
+
+      // Read the contents of the files at the given file paths
+      val options: List[String] = filePaths.flatMap { filePath =>
+        val source = scala.io.Source.fromFile(filePath)
+        source.getLines().toList
```

Comment:
This creates a leak because the source is never closed. Also, it's an inefficient way to read a file. Better to use something like `ZIO.readFile`.